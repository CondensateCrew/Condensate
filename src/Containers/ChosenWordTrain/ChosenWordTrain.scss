@import "../../css/variables.scss";

.chosen-train {
  display: -webkit-flex;
  display: -ms-flex;
  display: flex;
  width: 80vw;
  p {
    background: $primary-gradient;
    box-shadow: $external-shadow;
    border-radius: $one-line-border-radius;
    padding: $one-line-padding-space;
    margin-left: $small-space;
  }
}
