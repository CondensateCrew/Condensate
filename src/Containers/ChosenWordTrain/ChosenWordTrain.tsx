import React from 'react';
import { useSelector } from 'react-redux';
import './ChosenWordTrain.scss';
import { AppStore } from 'interfaces';

const ChosenWordTrain: React.FC = () => {
  const { chosenWords } = useSelector((store: AppStore) => ({
    chosenWords: store.chosenWords
  }));

  const blocks = chosenWords.map((word: string) => (
    <p key={word}>{word}</p>
  ));

  return (
    <section className="chosen-train">{ blocks }</section>
  )
}

export default ChosenWordTrain;
