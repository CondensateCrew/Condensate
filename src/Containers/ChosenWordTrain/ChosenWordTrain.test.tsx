import React from 'react';
import ChosenWordTrain from './ChosenWordTrain';
import { shallow } from 'enzyme';

jest.mock('react-redux', () => ({
  useSelector: () => ['Temp']
}));

describe('ChosenWordTrain component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<ChosenWordTrain />);
    expect(wrapper).toMatchSnapshot();
  });
});
