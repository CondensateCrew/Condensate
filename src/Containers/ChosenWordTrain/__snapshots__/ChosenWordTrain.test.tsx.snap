// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChosenWordTrain component should match the snapshot 1`] = `
<section
  className="chosen-train"
>
  <p
    key="Temp"
  >
    Temp
  </p>
</section>
`;
